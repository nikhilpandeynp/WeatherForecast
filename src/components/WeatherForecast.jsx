import React, { Component } from 'react';

export default class WeatherForecast extends Component {
    constructor(props) {
        super(props);
        this.state = {
            weatherForecast: this.props.weatherForecast
        };
    }

    render() {
        if (this.state.weatherForecast) {
            return <table className="table">
                <thead>
                    <tr>
                        {
                            this.state.weatherForecast.map((forecast, index) => {
                                let date = forecast.date;
                                let dateString = date.getDate().toString() + '/' + (date.getMonth() + 1).toString() + '/' + date.getFullYear().toString()
                                return <th scope="col" key={index}>{dateString}</th>
                            })
                        }
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        {
                            this.state.weatherForecast.map((forecast, index) => {
                                return <td key={index}>{forecast.temperature}</td>
                            })
                        }
                    </tr>
                </tbody>
            </table>
        } else {
            return <div></div>
        }
    }
}