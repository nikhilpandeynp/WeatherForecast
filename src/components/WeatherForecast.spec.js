import renderer from 'react-test-renderer';
import React from 'react';
import WeatherForecast from './WeatherForecast';

it('should render without weather forecast', () => {
    let component = renderer.create(<WeatherForecast />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render with weather forecast', () => {
    let props = {
        weatherForecast: [{
            date: new Date(),
            temperature: 31
        }]
    }
    let component = renderer.create(<WeatherForecast {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});