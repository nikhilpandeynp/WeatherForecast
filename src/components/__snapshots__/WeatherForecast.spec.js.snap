// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with weather forecast 1`] = `
<table>
  <thead>
    <tr>
      <th>
        16/5/2018
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        31
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should render without weather forecast 1`] = `<div />`;
